### PR TITLE
Port ExtensionUpgrade component

### LINKED ISSUE #60: Port over Extension Upgrade
Port over the `ExtensionUpgrade` component rendered by `ColonyHomeLayout`.

### PR DESCRIPTION
**Pull request by Julianahlk**
## Description

This PR Ports the `ExtensionUpgrade` component rendered by ColonyHomeLayout and `oneTxMustBeUpgraded`.

Resolves #60

### CODE REVIEW

File: src/components/common/ColonyHome/ExtensionUpgrade/ExtensionUpgrade.tsx
Comment (jakubcolony) [inline src/components/common/ColonyHome/ExtensionUpgrade/ExtensionUpgrade.tsx:12]: Should be higher up

File: src/utils/checks/oneTxMustBeUpgraded.ts
Comment (jakubcolony) [inline src/utils/checks/oneTxMustBeUpgraded.ts:1]: @rdig I didn't know this constant exists, how does it relate to storing the most recent version available in the network in the DB and exposing them on extension data via `availableVersion` field?
In the dapp it was yet something different, we were checking for a specific OneTxPayment version...

Comment (jakubcolony) [inline src/utils/checks/oneTxMustBeUpgraded.ts:1]: On a different note, would it make sense to put this in the existing `utils/extensions` file or to create a separate subfolder for extensions utils rather than scatter them around?

**Review by jakubcolony — COMMENTED** [review]
Just commenting for now, until I get an answer from Raul on how we should check for extensions version...

File: src/utils/checks/oneTxMustBeUpgraded.ts
Comment (Julianahlk) [reply src/utils/checks/oneTxMustBeUpgraded.ts]: Sure, since `utils/extension` is small right now, I moved `oneTxMustBeUpgraded` into that file. I think we can create a subfolder later if necessary.

Comment (jakubcolony) [reply src/utils/checks/oneTxMustBeUpgraded.ts]: Good point @willm30. Raul wanted to keep the versions in the database. I guess we'll find out once he's back

Comment (Julianahlk) [reply src/utils/checks/oneTxMustBeUpgraded.ts]: @rdig any thoughts on the comments above?

Comment (Julianahlk) [reply src/utils/checks/oneTxMustBeUpgraded.ts]: @jakubcolony @willm30 discussed with Raul and the extension version should come from the DB but for now we can leave it as is until #163 is merged. I left a comment in the code.

Comment (willm30) [reply src/utils/checks/oneTxMustBeUpgraded.ts]: @Julianahlk #163 is keeping track of the Colony version, you should be able to get the extension version from the db in this branch

Comment (Julianahlk) [reply src/utils/checks/oneTxMustBeUpgraded.ts]: Oh okay cool. I will look into it. Thanks 👍

Comment (Julianahlk) [reply src/utils/checks/oneTxMustBeUpgraded.ts]: @jakubcolony and @willm30 I used `GetCurrentExtensionVersion` query as suggested. Can you both please take a look again? Thank you.

File: src/hooks/useOneTxMustBeUpgraded.ts
Comment (Julianahlk) [inline src/hooks/useOneTxMustBeUpgraded.ts:24]: Using isInstalledExtensionData so that this notifies the user to upgrade both if the extension is deprecated or enabled.

File: src/hooks/useOneTxMustBeUpgraded.ts
Comment (willm30) [inline src/hooks/useOneTxMustBeUpgraded.ts:7]: Type definition needs to include the `| null` since that's how `extensionData` is defined in `ExtensionUpgrade`

File: src/components/common/ColonyHome/ExtensionUpgrade/ExtensionUpgrade.tsx
Comment (willm30) [inline src/components/common/ColonyHome/ExtensionUpgrade/ExtensionUpgrade.tsx:43]: Just checking this is how it works in the Dapp? That the banner only appears if the extension has been installed?

Is it not also possible to upgrade an extension that hasn't been installed yet? I.e. so that you install the most up to date version?

Comment (willm30) [reply src/components/common/ColonyHome/ExtensionUpgrade/ExtensionUpgrade.tsx:43]: Should also include the `| null` typing from `extensionData`

**Review by willm30 — CHANGES_REQUESTED** [review]
If you set `mustUpgrade` to true (line 46 of `ExtensionUpgrade`), when clicking the "Actions" navigation item, we get an error:

appcrash

The `ExtensionUpgrade` component presumably needs to be generalised so it can be shared across extensions? I.e. if the voting extension needs upgrading, we should be able to use the same component. Currently it's hardcoded for `OneTx`. 

And I'm seeing the Upgrade button even without the banner telling me to upgrade. I think these should be synchronised (i.e. we shouldn't see the upgrade button if there's nothing to upgrade to). 

upgrade-no-banner

**Review by jakubcolony — CHANGES_REQUESTED** [review]
I feel like we should rename this component. It suggests it covers upgrading of (possibly any) extension, while in reality it only seems to be an alert informing of an outdated version of OneTxPayment (and no other extension).

I thought it would be helpful to provide some testing instructions (which would involve manually updating the current version of OneTxPayment in the db to 4), but this is currently blocked by the bug I fixed in #163. If it gets merged before, I will come back and provide the instructions

File: src/components/common/ColonyHome/ExtensionUpgrade/ExtensionUpgrade.tsx
Comment (Julianahlk) [reply src/components/common/ColonyHome/ExtensionUpgrade/ExtensionUpgrade.tsx:40]: @jakubcolony maybe the variable name doesn't make sense but based on the dapp it seems the purpose of isExtensionDetailsRoute is to determine if the user is currently on the details p[age for OneTxPayment to determine whether to show a button to navigate to OneTxPayment details page

File: src/hooks/useOneTxMustBeUpgraded.ts
Comment (Julianahlk) [reply src/hooks/useOneTxMustBeUpgraded.ts]: Removed useOneTxMustBeUpgraded so should no longer be relevant

**Comment by Julianahlk** [comment]
> If you set `mustUpgrade` to true (line 46 of `ExtensionUpgrade`), when clicking the "Actions" navigation item, we get an error:
> 
This should be resolved.

> The `ExtensionUpgrade` component presumably needs to be generalised so it can be shared across extensions? I.e. if the voting extension needs upgrading, we should be able to use the same component. Currently it's hardcoded for `OneTx`.

@willm30 My assumption is that OneTx is a required extension therefore it must be upgraded. Are you saying that the other extensions have the same upgrade requirement?

> And I'm seeing the Upgrade button even without the banner telling me to upgrade. I think these should be synchronised (i.e. we shouldn't see the upgrade button if there's nothing to upgrade to).
> 
The upgrade button wasn't in the scope of this PR so this could be an existing issue in the extensions feature? @jakubcolony

**Comment by jakubcolony** [comment]
>The upgrade button wasn't in the scope of this PR so this could be an existing issue in the extensions feature?

That sounds right, I did some fixes to the button in #158 but it probably needs another look when the entire branch is more or less ready

**Comment by willm30** [comment]
> My assumption is that OneTx is a required extension therefore it must be upgraded. Are you saying that the other extensions have the same upgrade requirement

I would have thought so. I.e. if we change the contracts of the voting reputation extension, it's going to require "upgrading". 
@jakubcolony Is that your understanding?

**Comment by jakubcolony** [comment]
@willm30 The upgrading functionality is already ported, this component's role was to alert people on an outdated OneTxPayment version - hence my request to change its name to be more specific. I checked with Raul this morning and there's no intentions to extend it to other extensions

**Review by willm30 — APPROVED** [review]
All looking good to me 👍

File: src/components/common/ColonyHome/ExtensionUpgrade/ExtensionUpgrade.tsx
Comment (jakubcolony) [reply src/components/common/ColonyHome/ExtensionUpgrade/ExtensionUpgrade.tsx]: @Julianahlk I think this is still outstanding. The check is one-line, so we shouldn't need a separate function making extra query calls

File: src/components/common/Extensions/ExtensionStatusBadge/ExtensionStatusBadge.tsx
Comment (jakubcolony) [inline src/components/common/Extensions/ExtensionStatusBadge/ExtensionStatusBadge.tsx:50]: Why? It should never be `null`/`undefined`

**Review by jakubcolony — APPROVED** [review]
LGTM!

**Pull request merged**
